vector/src/main/java/fr/gouv/tchap/features/platform/TchapGetPlatformTask.kt

```diff
@@ -76,37 +74,22 @@ class TchapGetPlatformTask @Inject constructor(
     }
 
     private fun buildIdServerUrls(): List<String> {
-        val identityService = activeSessionHolder.getSafeActiveSession()?.identityService()
-        val idServerUrls: MutableList<String> = ArrayList()
+        val idServerUrls = mutableListOf<String>()
 
         // Consider first the current identity server if any.
-        val currentIdServerUrl = identityService?.getCurrentIdentityServerUrl()
+        val currentIdServerUrl = activeSessionHolder.getSafeActiveSession()?.identityService()?.getCurrentIdentityServerUrl()
         if (currentIdServerUrl != null) {
             idServerUrls.add(currentIdServerUrl)
         }
 
-        // Add randomly the preferred known ISes
-        var currentHosts: MutableList<String> = mutableListOf(*context.resources.getStringArray(R.array.preferred_identity_server_names))
-        while (currentHosts.isNotEmpty()) {
-            val index = Random().nextInt(currentHosts.size)
-            val host: String = currentHosts.removeAt(index)
-            val idServerUrl = context.getString(R.string.server_url_prefix) + host
-            if (currentIdServerUrl == null || idServerUrl != currentIdServerUrl) {
-                idServerUrls.add(idServerUrl)
-            }
-        }
-
-        // Add randomly the other known ISes
-        currentHosts = mutableListOf(*context.resources.getStringArray(R.array.identity_server_names))
-        while (currentHosts.isNotEmpty()) {
-            val index = Random().nextInt(currentHosts.size)
-            val host: String = currentHosts.removeAt(index)
-            val idServerUrl = context.getString(R.string.server_url_prefix) + host
-            if (currentIdServerUrl == null || idServerUrl != currentIdServerUrl) {
-                idServerUrls.add(idServerUrl)
-            }
-        }
+        // Compute list randomly from the preferred known ISes
+        val knownHosts = context.resources.getStringArray(R.array.preferred_identity_server_names).toList().shuffled()
```

Comment:
I like your suggestion. Just one remark. Could you replace toList() by toSet() ?